### PR TITLE
feat(@ciscospark/i-p-wdm): switch server when current server fails

### PR DESCRIPTION
**Pull request by tran2**
add server error interceptor to switch server url when a REST request
fails

### CODE REVIEW

File: packages/node_modules/@ciscospark/internal-plugin-wdm/src/interceptors/server-error.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+              success: false
+            },
+            tags: {
+              action: `failed`,
+              error: reason.message,
+              url: options.uri
+            }
+          });
+
+          return this.spark.internal.device.markUrlFailedAndGetNew(options.uri)
```
Comment (ianwremmel) [inline packages/node_modules/@ciscospark/internal-plugin-wdm/src/interceptors/server-error.js]: Should this request be replayed instead of failed?

Comment (tran2) [reply packages/node_modules/@ciscospark/internal-plugin-wdm/src/interceptors/server-error.js]: Do you mean retry this same request with the new server?

Comment (ianwremmel) [reply packages/node_modules/@ciscospark/internal-plugin-wdm/src/interceptors/server-error.js]: yea

**Comment by good-joe** [comment]
This PR should be closed, replaced by a nearly identical PR from jofuhrma

**Pull request closed without merging**